### PR TITLE
feat(pointcloud_preprocessor): calculate visibility score in ring outlier filter

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

calculate visibility score based on oulier pointcloud.
This feature is disabled by default with `publish_outlier_pointcloud`

## Tests performed

1. launch logging_simulator with following commnad
```
ros2 launch autoware_launch logging_simulator.launch.xml map_path:=/path_to_your_map vehicle_model:=lexus sensor_model:=aip_xx1 sensing:=true
```

2. Enabling this feature (for left lidar for example) with following command
```
ros2 param set /sensing/lidar/top/ring_outlier_filter publish_excluded_points true
```

3. play rosbag 
```
ros2 bag play --clock -r 0.5 /path_to_your_rosbag
```

4. output visibility score in the terminal
```
ros2 topic echo /sensing/lidar/top/ring_outlier_filter/debug/visibility
```

get the following result

```
---
stamp:
  sec: 1707114299
  nanosec: 200355584
data: 0.6482204794883728
---
stamp:
  sec: 1707114299
  nanosec: 300320000
data: 0.6338975429534912
---

```

## Effects on system behavior

not applicable

## Interface changes

not applicable

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
Can you add an algorithm summary to the readme?

**Comment by kyoichi-sugahara** [comment]
@yukkysaito 
Updated document in https://github.com/autowarefoundation/autoware.universe/pull/7011/commits/c61dab92e5f602cc1afe6823fb6f55a786a98e5e

**Comment by knzo25** [comment]
@vividf 
I will be reviewing:
https://github.com/autowarefoundation/autoware.universe/pull/7011
https://github.com/autowarefoundation/autoware.universe/pull/6746

So please review this one in the meantime

File: sensing/pointcloud_preprocessor/docs/ring-outlier-filter.md [hunk 1]
```diff
@@ -10,6 +10,40 @@ A method of operating scan in chronological order and removing noise based on th
 
 ![ring_outlier_filter](./image/outlier_filter-ring.drawio.svg)
 
+Another feature of this node is that it calculates visibility score based on outlier pointcloud and publish score as a topic.
+
+### visibility score calculation algorithm
+
+The pointcloud is divided into vertical bins (rings) and horizontal bins (azimuth divisions).
+The algorithm starts by splitting the input point cloud into separate rings based on the ring value of each point. Then, for each ring, it iterates through the points and calculates the frequency of points within each horizontal bin. The frequency is determined by incrementing a counter for the corresponding bin based on the point's azimuth value.
```
Comment (vividf) [inline sensing/pointcloud_preprocessor/docs/ring-outlier-filter.md:18]: point cloud -> pointcloud

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**